crates/relayer/src/middleware.rs

```diff
@@ -0,0 +1,59 @@
+use std::sync::Arc;
+use std::time::Instant;
+
+use axum::extract::Request;
+use axum::middleware::Next;
+use axum::response::IntoResponse;
+
+use crate::metrics::RelayerMetrics;
+
+pub async fn track_metrics(
+    metrics: Option<Arc<RelayerMetrics>>,
+    req: Request,
+    next: Next,
+) -> impl IntoResponse {
+    let Some(metrics) = metrics else {
+        // No metrics available, just pass through
+        return next.run(req).await;
+    };
+
+    let path = req.uri().path();
```

Comment:
This is a potential target for a cardinality explosion. `req.uri().path()` returns the exact string sent by the client. You can overload and crash the relayer by sending requests to random paths - e.g. `/other1, /other2/, etc.`
This is because Prometheus creates a new memory entry for every unique path.
Instead, please fix it to maintain instead the list of allowed paths, merging all the other stuff into another path. Could be done in something like that (please verify on your side):

```suggestion
    let raw_path = req.uri().path();
    
    // Normalize path to prevent cardinality explosion.
    let path = match raw_path {
        "/" => "/",
        "/health" => "/health",
        "/metrics" => "/metrics",
        _ => "other", 
    };
```